### PR TITLE
Validation start. Modifiers update.

### PR DESCRIPTION
**Pull request by DovydasNavickas**
The PR is still somewhat work-in-progress, but things already work here. Let's do the clean-up later on.

### CODE REVIEW

File: apps/demo/src/length-validator.tsx [hunk 1]
```diff
@@ -0,0 +1,45 @@
+import { useValidator } from "@reactway/forms/src";
```
Comment (MartynasZilinskas) [inline apps/demo/src/length-validator.tsx]: import location.

File: apps/demo/src/length-validator.tsx [hunk 2]
```diff
@@ -0,0 +1,45 @@
+import { useValidator } from "@reactway/forms/src";
+import { useCallback } from "react";
```
Comment (MartynasZilinskas) [inline apps/demo/src/length-validator.tsx]: React import should be at the top.

File: apps/demo/src/length-validator.tsx [hunk 3]
```diff
@@ -0,0 +1,45 @@
+import { useValidator } from "@reactway/forms/src";
+import { useCallback } from "react";
+import { ValidationResult } from "@reactway/forms-core/src";
```
Comment (MartynasZilinskas) [inline apps/demo/src/length-validator.tsx]: import location.

File: apps/demo/src/length-validator.tsx [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+    tooLong: string;
+}
+
+export interface LengthValidatorProps {
+    min: number;
+    max?: number;
+    messages: LengthValidatorMessages;
+}
+
+export const LengthValidator: React.FC<LengthValidatorProps> = (props): null => {
```
Comment (MartynasZilinskas) [inline apps/demo/src/length-validator.tsx]:
```
export const LengthValidator = (props: LengthValidatorProps): null => {
```

File: apps/demo/src/length-validator.tsx [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+
+export interface LengthValidatorMessages {
+    tooShort: string;
+    tooLong: string;
+}
+
+export interface LengthValidatorProps {
+    min: number;
+    max?: number;
+    messages: LengthValidatorMessages;
```
Comment (MartynasZilinskas) [inline apps/demo/src/length-validator.tsx]: errorMessages?

File: packages/forms/src/validators/base.tsx [hunk 6]
```diff
@@ -0,0 +1,11 @@
+import { Validator } from "@reactway/forms-core";
+import { useValidator } from "../use-validator";
+
+export interface BaseValidatorProps<TValue> {
+    validator: Validator<TValue>;
+}
+
+export const BaseValidator = <TValue,>(props: BaseValidatorProps<TValue>): null => {
```
Comment (MartynasZilinskas) [inline packages/forms/src/validators/base.tsx:8]:
```
<TValue,>
```
`,` ?

**Review by MartynasZilinskas — CHANGES_REQUESTED** [review]

File: apps/demo/src/length-validator.tsx (hunk 4 above)
Comment (DovydasNavickas) [reply apps/demo/src/length-validator.tsx]: So it's defaulting the props inside the component then 😄
https://twitter.com/dan_abramov/status/1133878326358171650

**Review by MartynasZilinskas — APPROVED** [review]

**Comment by DovydasNavickas** [comment]
The PR is still somewhat work-in-progress, but things already work here. Let's do the clean-up later on.

**Pull request merged**